public/app/features/org/NewOrgPage.tsx

```diff
@@ -30,13 +24,21 @@ const validateOrg = async (orgName: string) => {
 
 interface PropsWithState {
   navModel: NavModel;
+  setUserOrganization: typeof setUserOrganization;
 }
 
 interface CreateOrgFormDTO {
   name: string;
 }
 
-export const NewOrgPage: FC<PropsWithState> = ({ navModel }) => {
+export const NewOrgPage: FC<PropsWithState> = ({ navModel, setUserOrganization }) => {
+  const createOrg = async (newOrg: { name: string }) => {
+    const result = await getBackendSrv().post('/api/orgs/', newOrg);
+
+    setUserOrganization(result.orgId);
+    window.location.href = getConfig().appSubUrl + '/org';
+  };
+
```

Comment:
do you think we can just move all the `createOrg` logic into the reducer? 🤔

Reply:
yeah...sounds like a good idea to me

Reply:
> move all the createOrg logic into the reducer

The reducer you mean here is all the logic in `org/state/action.ts` right? You know these terms can become a bit confusing sometimes 😅